File: images/lir_node/lir_node/modules/frr/frr_manager.py
import os
from modules.config import env_loader as elm


def start_frr():
    """
    启动 frr
    :return:
    """
    enableFrr = elm.env_loader.enable_frr
    if enableFrr == "true":
        containerName = elm.env_loader.container_name
        sourceFilePath = f"/configuration/{containerName}/route/frr.conf"
        destFilePath = f"/etc/frr/frr.conf"
        copy_command = f"cp {sourceFilePath} {destFilePath}"
        start_frr_command = "service frr start"
        os.system(copy_command)
        os.system(start_frr_command)
        print("start frr", flush=True)
    else:
        print("not start frr", flush=True)

File: images/ground_station/ground_station/modules/frr/frr_manager.go
package frr

import (
	"bytes"
	"fmt"
	"os"
	"os/exec"
)

// CopyFrrConfigurationFile 进行 FRR 配置文件的拷贝
func CopyFrrConfigurationFile() {
	containerName := os.Getenv("CONTAINER_NAME")

	sourceFilePath := fmt.Sprintf("/configuration/%s/route/frr.conf", containerName)

	destFilePath := "/etc/frr/frr.conf"

	cmd := exec.Command("cp", sourceFilePath, destFilePath)
	var out bytes.Buffer
	cmd.Stdout = &out

	err := cmd.Run()
	if err != nil {
		fmt.Println(err)
	}
}

// StartFrr 执行启动 frr 的命令
func StartFrr() {
	cmd := exec.Command("service", "frr", "start")
	output, err := cmd.CombinedOutput()
	if err != nil {
		panic(err)
	}
	fmt.Println(string(output))
}

// Start 拷贝 frr + 启动 frr
func Start() {
	enableFrr := os.Getenv("ENABLE_FRR")
	if enableFrr == "true" {
		fmt.Println("start frr")
		CopyFrrConfigurationFile()
		StartFrr()
	} else {
		fmt.Println("not start frr")
	}
}
